js/src/views/Signer/components/RequestOrigin/requestOrigin.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+
+import IdentityIcon from '~/ui/IdentityIcon';
+
+import styles from './requestOrigin.css';
+
+export default class RequestOrigin extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    origin: PropTypes.oneOfType([
```

Comment:
I'd prefer having two props, e.g. `type` and `origin` here, one containing `rpc`/`ipc`/etc and one containing the data necessary to display it. Better to read and cleaner than passing an object as a prop.